indra_gpt/pipelines/statement_extraction_pipeline.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import pickle
+from pathlib import Path
+
+class StatementExtractionPipeline:
+    def __init__(self, pre_processor, generator, post_processor):
+        """
+        Initializes the pipeline with preprocessing, generation, and postprocessing components.
+        """
+        self.pre_processor = pre_processor
+        self.generator = generator
+        self.post_processor = post_processor
+        self.logger = logging.getLogger(__name__)
+
+    def run(self):
+        """
+        Executes the entire extraction pipeline: preprocessing → generation → postprocessing.
```

Comment:
We should try to avoid less common characters like '→' , since we want to minimize potential issues that could arise if we need to parse the text in the future.
```suggestion
        Executes the entire extraction pipeline: preprocessing, generation and postprocessing.
```